File: app/src/main/java/org/neidhardt/dynamicsoundboard/soundmanagement/model/SoundsManagerUtil.java
package org.neidhardt.dynamicsoundboard.soundmanagement.model;

import org.neidhardt.dynamicsoundboard.dao.SoundLayout;
import org.neidhardt.dynamicsoundboard.mediaplayer.EnhancedMediaPlayer;
import org.neidhardt.dynamicsoundboard.soundlayoutmanagement.model.SoundLayoutsAccess;

import java.util.List;
import java.util.Map;
import java.util.Set;

/**
 * File created by eric.neidhardt on 15.06.2015.
 */
public class SoundsManagerUtil
{
	static final String DB_SOUNDS_DEFAULT = "org.neidhardt.dynamicsoundboard.storage.SoundManagerFragment.db_sounds";
	static final String DB_SOUNDS_PLAYLIST_DEFAULT = "org.neidhardt.dynamicsoundboard.storage.SoundManagerFragment.db_sounds_playlist";

	private static final String DB_SOUNDS = "db_sounds";
	private static final String DB_SOUNDS_PLAYLIST = "db_sounds_playlist";

	static String getDatabaseNameSounds(SoundLayoutsAccess soundLayoutsAccess)
	{
		SoundLayout activeLayout = soundLayoutsAccess.getActiveSoundLayout();
		if (activeLayout.isDefaultLayout())
			return DB_SOUNDS_DEFAULT;
		String baseName = activeLayout.getDatabaseId();
		return baseName + DB_SOUNDS;
	}

	static String getDatabaseNamePlayList(SoundLayoutsAccess soundLayoutsAccess)
	{
		SoundLayout activeLayout = soundLayoutsAccess.getActiveSoundLayout();
		if (activeLayout.isDefaultLayout())
			return DB_SOUNDS_PLAYLIST_DEFAULT;
		String baseName = activeLayout.getDatabaseId();
		return baseName + DB_SOUNDS_PLAYLIST;
	}

	public static EnhancedMediaPlayer searchInListForId(String playerId, List<EnhancedMediaPlayer> sounds)
	{
		if (sounds == null)
			return null;
		for (EnhancedMediaPlayer player : sounds)
		{
			if (player.getMediaPlayerData().getPlayerId().equals(playerId))
				return player;
		}
		return null;
	}

	public static EnhancedMediaPlayer searchInMapForId(String playerId, Map<String, List<EnhancedMediaPlayer>> sounds)
	{
		Set<String> soundSheets = sounds.keySet();
		for (String soundSheet : soundSheets)
		{
			List<EnhancedMediaPlayer> playersInSoundSheet = sounds.get(soundSheet);
			EnhancedMediaPlayer player = searchInListForId(playerId, playersInSoundSheet);
			if (player != null)
				return player;
		}
		return null;
	}
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/soundmanagement/model/SoundsManagerUtil.kt
package org.neidhardt.dynamicsoundboard.soundmanagement.model

import org.neidhardt.dynamicsoundboard.mediaplayer.MediaPlayerController
import org.neidhardt.dynamicsoundboard.soundlayoutmanagement.model.ISoundLayoutManager
import org.neidhardt.dynamicsoundboard.soundlayoutmanagement.model.activeLayout

/**
 * File created by eric.neidhardt on 15.06.2015.
 */
internal val DB_SOUNDS_DEFAULT = "org.neidhardt.dynamicsoundboard.storage.SoundManagerFragment.db_sounds"
internal val DB_SOUNDS_PLAYLIST_DEFAULT = "org.neidhardt.dynamicsoundboard.storage.SoundManagerFragment.db_sounds_playlist"

private val DB_SOUNDS = "db_sounds"
private val DB_SOUNDS_PLAYLIST = "db_sounds_playlist"

internal fun getDatabaseNameSounds(manager: ISoundLayoutManager): String
{
	val activeLayout = manager.soundLayouts.activeLayout
	if (activeLayout.isDefaultLayout)
		return DB_SOUNDS_DEFAULT
	val baseName = activeLayout.databaseId
	return baseName + DB_SOUNDS
}

internal fun getDatabaseNamePlayList(manager: ISoundLayoutManager): String
{
	val activeLayout = manager.soundLayouts.activeLayout
	if (activeLayout.isDefaultLayout)
		return DB_SOUNDS_PLAYLIST_DEFAULT
	val baseName = activeLayout.databaseId
	return baseName + DB_SOUNDS_PLAYLIST
}

internal fun searchInListForId(playerId: String, sounds: List<MediaPlayerController>): MediaPlayerController?
{
	return sounds.firstOrNull { player -> player.mediaPlayerData.playerId == playerId }
}

internal fun searchInMapForId(playerId: String, sounds: Map<String, List<MediaPlayerController>>): MediaPlayerController?
{
	val soundSheets = sounds.keys
	return soundSheets
			.map { sounds[it] }
			.map { searchInListForId(playerId, it.orEmpty()) }
			.firstOrNull { it != null }
}

